LinkedBlockingQueueSerializer fails to deserialize default unbounded queues (checkReadableBytes(capacity))

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/fory/issues) and found no similar issues.


### Version

Fory 1.2.0, Java 23.

### Component(s)

Java

### Minimal reproduce step

  Fory 1.2.0, Java 23.

  Fory fory = Fory.builder()
      .withLanguage(Language.JAVA)
      .withRefTracking(false)
      .withCompatibleMode(CompatibleMode.SCHEMA_CONSISTENT)
      .requireClassRegistration(false)
      .build();
  LinkedBlockingQueue<String> queue = new LinkedBlockingQueue<>();
  queue.add("a");
  queue.add("b");
  byte[] bytes = fory.serialize(queue);
  fory.deserialize(bytes);

  Same thing with a bounded queue works:

  LinkedBlockingQueue<String> bounded = new LinkedBlockingQueue<>(16);
  bounded.add("a");
  fory.deserialize(fory.serialize(bounded)); // fine

### What did you expect to see?

serialize + deserialize round-trip should work for a default new LinkedBlockingQueue<>(), same as other collection types (ArrayDeque, ArrayList, etc.).

### What did you see instead?

Serialize succeeds. Deserialize throws:

  org.apache.fory.exception.DeserializationException: Failed to deserialize input
  Caused by: java.lang.IndexOutOfBoundsException: readerIndex(9) + length(2147483532) exceeds size(124)
      at org.apache.fory.serializer.collection.CollectionSerializers$LinkedBlockingQueueSerializer.newCollection(CollectionSerializers.java:997)

My guess: newCollection reads capacity off the wire and passes it to checkReadableBytes(capacity). For an unbounded queue that capacity is ~Integer.MAX_VALUE, so it blows up. Bounded new LinkedBlockingQueue<>(16) doesn't hit this.

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/java/fory-core/src/main/java/org/apache/fory/serializer/collection/CollectionSerializers.java
+++ b/java/fory-core/src/main/java/org/apache/fory/serializer/collection/CollectionSerializers.java
@@ -994,7 +994,6 @@ public class CollectionSerializers {
       setNumElements(numElements);
       int capacity = buffer.readVarUInt32Small7();
       checkBoundedQueueCapacity(numElements, capacity);
-      buffer.checkReadableBytes(capacity);
       LinkedBlockingQueue queue = new LinkedBlockingQueue<>(capacity);
       readContext.reference(queue);
       return queue;

--- a/java/fory-core/src/test/java/org/apache/fory/serializer/collection/CollectionSerializersTest.java
+++ b/java/fory-core/src/test/java/org/apache/fory/serializer/collection/CollectionSerializersTest.java
@@ -1159,6 +1159,7 @@ public class CollectionSerializersTest extends ForyTestBase {
       queue.add(2);
       LinkedBlockingQueue<Integer> deserialized = serDe(fory, queue);
       assertEquals(new ArrayList<>(deserialized), new ArrayList<>(queue));
+      assertEquals(deserialized.remainingCapacity() + deserialized.size(), Integer.MAX_VALUE);
     }
   }
 

--- a/java/fory-core/src/test/java/org/apache/fory/serializer/collection/CollectionSerializersTest.java
+++ b/java/fory-core/src/test/java/org/apache/fory/serializer/collection/CollectionSerializersTest.java
@@ -1153,6 +1153,13 @@ public class CollectionSerializersTest extends ForyTestBase {
       assertEquals(new ArrayList<>(deserialized), new ArrayList<>(queue));
       assertEquals(deserialized.remainingCapacity() + deserialized.size(), 3);
     }
+    {
+      LinkedBlockingQueue<Integer> queue = new LinkedBlockingQueue<>();
+      queue.add(1);
+      queue.add(2);
+      LinkedBlockingQueue<Integer> deserialized = serDe(fory, queue);
+      assertEquals(new ArrayList<>(deserialized), new ArrayList<>(queue));
+    }
   }
 
   @Test

